R/LSbinaryclassification.R

```diff
@@ -0,0 +1,1291 @@
+  if(!is.null(plotsContainer[["plotIconPlot"]]) ) return()
+
+  plotsContainer[["plotIconPlot"]] <-
+    createJaspPlot(title        = gettext("Icon plot"),
+                   dependencies = "plotIconPlot",
+                   position     = position,
+                   aspectRatio  = 1,
+                   width        = 600,
+                   height       = 400
+    )
+
+  if(ready) plotsContainer[["plotIconPlot"]]$plotObject <-
+    .bcFillPlotIconPlot(results, dataset, options)
+
+}
+
+.bcFillPlotIconPlot <- function(results, dataset, options) {
+  UseMethod(".bcFillPlotIconPlot")
+}
+
+.bcFillPlotIconPlot.default <- function(results, dataset, options) {
+
+  data <- expand.grid(cond = gettext(c("Positive", "Negative")),
+                      test = gettext(c("Positive", "Negative")),
+                      KEEP.OUT.ATTRS = FALSE)
+  data$out <- factor(
+    gettext(c("True positive", "False positive", "False negative", "True negative")),
+    levels = gettext(c("True positive", "False positive", "False negative", "True negative"))
+  )
+  data$prop <- summary(results, ciLevel = options[["ciLevel"]])[c("truePositive", "falsePositive", "falseNegative", "trueNegative"), "estimate"]
+
+  if(any(data$prop < 1e-4)) {
+    npoints <- 1e6
+    xside <- yside <- 1e3
+  } else if(any(data$prop < 1e-2)) {
+    npoints <- 1e4
+    xside <- yside <- 1e2
+  } else {
+    npoints <- 1e2
+    xside <- yside <- 10
+  }
+
+  data$n <- round(npoints*data$prop, digits = 0)
+  data <- data.frame(
+    outcome = rep(data$out, data$n),
+    x = rep(c(1:xside,xside:1), times = yside/2),
+    y = rep(seq_len(yside), each = xside),
+    image = system.file("icons", "person.svg", package = "jaspLearnBayes")
+  )
+
+  plot <- ggplot2::ggplot(data=data, mapping = ggplot2::aes(x=x,y=y))
+
+  if(npoints <= 1e2) {
+    plot <- plot + ggimage::geom_image(mapping = ggplot2::aes(col=outcome,image=image)) +
+      ggplot2::scale_color_manual(name = "", values = c("darkgreen", "darkorange", "red", "steelblue"), labels = gettext(c("True positive", "False positive", "False negative", "True negative")))
+  } else {
+    plot <- plot + ggplot2::geom_tile(mapping = ggplot2::aes(fill=outcome)) +
+      ggplot2::scale_fill_manual (name = "", values = c("darkgreen", "darkorange", "red", "steelblue"), labels = gettext(c("True positive", "False positive", "False negative", "True negative")))
+  }
+
+  plot <- plot +
+    ggplot2::coord_fixed(ratio = xside/yside) +
+    ggplot2::ylab(NULL) + ggplot2::xlab(NULL) +
+    ggplot2::scale_x_discrete(labels = NULL, breaks = NULL) +
+    ggplot2::scale_y_discrete(labels = NULL, breaks = NULL)
+
+  plot <- jaspGraphs::themeJasp(plot, legend.position = "right")
+
+  return(plot)
+}
+
+## ROC curve plot ----
+.bcPlotROC <- function(results, plotsContainer, dataset, options, ready, position) {
+  if( isFALSE(options[["plotROC"]])     ) return()
+  if(!is.null(plotsContainer[["plotROC"]]) ) return()
+
+  plotsContainer[["plotROC"]] <-
```

Comment:
No necessary, but an idea might be to add something like `ggrepel::geom_text_repel(data = data.frame(x = 0, y = 1), ggplot2::aes(label= gettext("Perfect separation"), x = x, y = y),
                                       nudge_x = -0.1, nudge_y = 0.05, xlim = c(0, 1), ylim = c(0, 1.1))` like in the ROC curve plot in the machine learning module:
![image](https://user-images.githubusercontent.com/25059399/116854406-ea3bee00-abf7-11eb-8c59-5441d46d9f5f.png)